tests/unit/python/foglamp/services/core/test_server.py

```diff
@@ -31,21 +38,140 @@
+            args, kwargs = audit_info_patch.call_args
+            assert 'FSTOP' == args[0]
+            assert None is args[1]
+
+        assert 1 == mocked__stop_scheduler.call_count
+        assert 1 == mocked_stop_microservices.call_count
+        assert 1 == mocked_stop_service_monitor.call_count
+        assert 1 == mocked_stop_rest_server.call_count
+        assert 1 == mocked_stop_storage.call_count
+        assert 1 == mocked__remove_pid.call_count
+
+    @pytest.mark.asyncio
+    @pytest.mark.skip
+    def test__start_core(self, mocker, loop):
+        async def return_async_value(val):
+            return val
+
+        mocker.patch.object(Server, "_make_core_app")
+
+        mock_start_storage = mocker.patch.object(Server, "_start_storage")
+        mock_start_storage.return_value = return_async_value('start storage')
+
+        mockedSC = MagicMock(spec=StorageClient)
+        mock_get_storage_client = mocker.patch.object(Server, "_get_storage_client")
+        mock_get_storage_client.return_value = return_async_value(mockedSC)
+
+        mockedCf = MagicMock(spec=ConfigurationManager(mockedSC))
+        mockedIntReg = MagicMock(spec=InterestRegistry)
+
+        mockedAuditLogger = MagicMock(spec=AuditLogger(mockedSC))
+
+        mock_start_scheduler = mocker.patch.object(Server, "_start_scheduler")
+        mock_start_scheduler.return_value = return_async_value('start scheduler')
+
+        mock_start_service_monitor = mocker.patch.object(Server, "_start_service_monitor")
+        mock_start_service_monitor.return_value = return_async_value('start monitor')
+
+        mock_rest_api_config = mocker.patch.object(Server, "rest_api_config")
+        mock_rest_api_config.return_value = return_async_value('set RET api config')
+
+        mock_service_config = mocker.patch.object(Server, "service_config")
+        mock_service_config.return_value = return_async_value('set service config')
+
+        def stop_me():
+            loop.stop()
+
+        m = MagicMock()
+        m.sockets[0].getsockname.return_value = ('127.0.0.1', 52300)
+
+        # patch pid
+        # set ssl false/ true
+
+        # def se(*args):
+        #     print(args[0])
+        #     m = MagicMock()
+        #     m.sockets[0].getsockname.return_value = ('127.0.0.1', 52300)
+        #     return m, "b"
+
+        # with patch.object(Server, "_register_core", return_value=None) as patch_registration:
+        with patch.object(Server, "_start_app", return_value=(m, "b")):
+            with patch.object(server._logger, "info", return_value=None) as logger_patch:
+                loop.call_later(2, Server.stop, loop)
+                Server._start_core(loop=loop)
+        # patch_registration.assert_called_once_with('0.0.0.0', 52300, 52300)
+        assert 6 == logger_patch.call_count
+        # calls = 'REST API Server started on %s://%s:%s', 'http', '127.0.0.1', 52300
+        # logger_patch.assert_has_calls(calls)
+
```

Comment:
add all pending tests stub for protected/ private def units, with `@pytest.mark.skip` for next PR.